From 2adeba3ef1f8cf283b0d3c3b9c4a0e009fc85f65 Mon Sep 17 00:00:00 2001
From: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
Date: Fri, 25 Mar 2022 14:36:33 +0530
Subject: [PATCH] hostapd: unable to config unsolicited broadcast Probe
 Response

Unsolicited broadcast Probe Response config is not working in hostapd,
this is because fils_discovery_min_int was used by mistake where
fils_discovery_max_int should have been used, latter is the one used to
decide whether FILS discovery is enabled or not.

Signed-off-by: Karthikeyan Kathirvel <quic_kathirve@quicinc.com>
---
 src/ap/ap_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index e91c042..4ceb480 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1427,7 +1427,7 @@ static int hostapd_config_check_bss(struct hostapd_bss_config *bss,
 #endif /* CONFIG_SAE_PK */
 
 #ifdef CONFIG_FILS
-	if (full_config && bss->fils_discovery_min_int &&
+	if (full_config && bss->fils_discovery_max_int &&
 	    bss->unsol_bcast_probe_resp_interval) {
 		wpa_printf(MSG_ERROR,
 			   "Cannot enable both FILS discovery and unsolicited broadcast Probe Response at the same time");
-- 
2.7.4

